### PR TITLE
609 Fix AssemblyLoadContext.Run

### PR DESCRIPTION
**Pull request by baseTwo**
Fix for 
* #609 

⚠️Previous PR #603  must be merged first

Loads and executes singleton libraries correctly in `AssemblyLoadContextExtensions.Run`
Unit test added `AssemblyLoadContextTest`

### CODE REVIEW

**Comment by baseTwo** [comment]
cc @and-hus I added your unit tests for running cql nested tuples from an assembly load context and serializing it in this PR

**Comment by and-hus** [comment]
> cc @and-hus I added your unit tests for running cql nested tuples from an assembly load context and serializing it in this PR

Great - thanks!
Just one remark. I see that in serialization test you are loading the tuple from the same assembly like:

```csharp
var obj = CqlNestedTupleTest_1_0_0.Instance.Result(ctx);
```

Be aware that I deliberately loaded and executed from an external assembly to ensure that whole information is passed across assembly boundaries - as this was the main issue with the initial implementation...

File: 0001-Add-unit-test-to-retrieve-tuple-across-assembly-boun.patch [hunk 1]
```diff
@@ -0,0 +1,351 @@
+From e430d95c96041334f7c17c64487d9b50697210bc Mon Sep 17 00:00:00 2001
```
Comment (and-hus) [inline 0001-Add-unit-test-to-retrieve-tuple-across-assembly-boun.patch]: I don't think that this patch file should be commited to the repo...
I assume it was added by accident.

Comment (baseTwo) [reply 0001-Add-unit-test-to-retrieve-tuple-across-assembly-boun.patch]: Well spotted

**Review by ewoutkramer — DISMISSED** [review]

**Comment by baseTwo** [comment]
> > cc @and-hus I added your unit tests for running cql nested tuples from an assembly load context and serializing it in this PR
> 
> Great - thanks! Just one remark. I see that in serialization test you are loading the tuple from the same assembly like:
> 
> ```cs
> var obj = CqlNestedTupleTest_1_0_0.Instance.Result(ctx);
> ```
> 
> Be aware that I deliberately loaded and executed from an external assembly to ensure that whole information is passed across assembly boundaries - as this was the main issue with the initial implementation...

Noted, I could add the unit test to load from external assembly to the one loading directly from a library

**Review by ewoutkramer — APPROVED** [review]

**Pull request merged**